src/azure-cli/azure/cli/command_modules/rdbms/_params.py

```diff
@@ -76,18 +75,19 @@ def _complex_params(command_group):
 
         with self.argument_context('{} server create'.format(command_group)) as c:
             c.argument('server_name', options_list=['--name', '-n'], arg_type=server_name_setter_arg_type)
-            c.argument('sku_name', options_list=['--sku-name'], required=True, help='The name of the sku. Follows the convention {pricing tier}_{compute generation}_{vCores} in shorthand. Examples: B_Gen5_1, GP_Gen5_4, MO_Gen5_16. ')
-            c.argument('administrator_login', required=True, arg_group='Authentication')
-            c.argument('administrator_login_password', required=True, arg_group='Authentication')
+            c.argument('sku_name', default='GP_Gen5_2', options_list=['--sku-name'],
+                       help='The name of the sku. Follows the convention {pricing tier}_{compute generation}_{vCores} in shorthand. Examples: B_Gen5_1, GP_Gen5_4, MO_Gen5_16. ')
+            c.argument('administrator_login', default=generate_username(), arg_group='Authentication')
+            c.argument('administrator_login_password', arg_group='Authentication')
 
             c.argument('backup_retention', type=int, options_list=['--backup-retention'], help='The number of days a backup is retained. Range of 7 to 35 days. Default is 7 days.', validator=retention_validator)
             c.argument('geo_redundant_backup', arg_type=get_enum_type(['Enabled', 'Disabled']), options_list=['--geo-redundant-backup'], help='Enable or disable geo-redundant backups. Default value is Disabled. Not supported in Basic pricing tier.')
-            c.argument('storage_mb', options_list=['--storage-size'], type=int, help='The storage capacity of the server (unit is megabytes). Minimum 5120 and increases in 1024 increments. Default is 51200.')
+            c.argument('storage_mb', default=51200, options_list=['--storage-size'], type=int, help='The storage capacity of the server (unit is megabytes). Minimum 5120 and increases in 1024 increments. Default is 51200.')
             c.argument('auto_grow', arg_type=get_enum_type(['Enabled', 'Disabled']), options_list=['--auto-grow'], help='Enable or disable autogrow of the storage. Default value is Enabled.')
             c.argument('infrastructure_encryption', arg_type=get_enum_type(['Enabled', 'Disabled']), options_list=['--infrastructure-encryption', '-i'], help='Add an optional second layer of encryption for data using new encryption algorithm. Default value is Disabled.')
             c.argument('assign_identity', options_list=['--assign-identity'], help='Generate and assign an Azure Active Directory Identity for this server for use with key management services like Azure KeyVault.')
 
-            c.argument('location', arg_type=get_location_type(self.cli_ctx), validator=get_default_location_from_resource_group)
+            c.argument('location', arg_type=get_location_type(self.cli_ctx))
```

Comment:
why not get default value from resource group?

Reply:
There are two cases for this :
The user can choose whether or not to provide the location. 
 - In case they do, we take up the location they provided. 
 - In case they don't - we look for it in the local context, else, we provision it in the default location (eastus for PG and westus2 for Mysql and MariaDB). When the user does not provide the resource group, we create the resource group as well. 
Having the above validator would mean that the user **always** has to pass the resource group from which the location will be determined. However, based on the current implementation, the user can choose **not** to provide any arguments and we will create it for them. They will just issue the command - az postgres server create and we will create the resource group and in the default location and use that location for any other action that follows.